src/main/java/com/powsybl/openloadflow/network/impl/LfShuntImpl.java

```diff
@@ -6,35 +6,119 @@
  */
 package com.powsybl.openloadflow.network.impl;
 
-import com.powsybl.iidm.network.ShuntCompensator;
+import com.powsybl.iidm.network.*;
 import com.powsybl.openloadflow.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import java.util.List;
-import java.util.Objects;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
  */
 public class LfShuntImpl extends AbstractElement implements LfShunt {
 
-    private final List<ShuntCompensator> shuntCompensators;
+    private static final Logger LOGGER = LoggerFactory.getLogger(LfShuntImpl.class);
 
-    private final double b;
+    private final List<ShuntCompensator> fixedShunts = new ArrayList<>();
+
+    private final List<ShuntCompensator> controllerShunts = new ArrayList<>();
+
+    private double b;
 
     protected LfBus bus;
 
+    private final List<ControllerLfShunt> controllerLfShunts = new ArrayList<>();
+
+    private double zb;
+
+    private static class ControllerLfShunt {
+
+        private double bAmplitude;
+
+        private Integer position;
+
+        private List<Double> sections;
+
+        private String id;
+
+        public ControllerLfShunt(List<Double> sections, Integer position, String id) {
+            this.sections = sections;
+            this.position = position;
+            double bMin = Math.min(sections.get(0), sections.get(sections.size() - 1));
+            double bMax = Math.max(sections.get(0), sections.get(sections.size() - 1));
+            this.bAmplitude = Math.abs(bMax - bMin);
+            this.id = id;
+        }
+
+        public List<Double> getSections() {
+            return this.sections;
+        }
+
+        public Integer getPosition() {
+            return this.position;
+        }
+
+        public void setPosition(Integer position) {
+            this.position = position;
+        }
+
+        public double getB() {
+            return this.sections.get(this.position);
+        }
+
+        public double getBAmplitude() {
+            return this.bAmplitude;
+        }
+
+        public String getId() {
+            return this.id;
+        }
+    }
+
     public LfShuntImpl(List<ShuntCompensator> shuntCompensators, LfNetwork network) {
         super(network);
         if (shuntCompensators.isEmpty()) {
             throw new IllegalArgumentException("Empty shunt compensator list");
         }
-        this.shuntCompensators = Objects.requireNonNull(shuntCompensators);
+        for (ShuntCompensator sc : shuntCompensators) {
```

Comment:
But looking at how you build ShuntImpl, there either only shunts controlling voltage or only shunts not controlling voltage?

Reply:
Yes, so I think that you are right I can simply the `LfShuntImpl` knowing that.